File: 2020_advent/Day_5-[Binary_Boarding]/python/exercise_2.py
input = open('input_data.txt', 'r')

def str_arr_replace(phrase, char_to_replace):
    for key, value in char_to_replace.items():
        phrase = phrase.replace(key, value)
    return phrase

airplane = [['.' for i in range(8)] for i in range(128)]
row = 0
col = 0
i = 0

for seat in input:
    board_data = str_arr_replace(seat, {'F':'0', 'B':'1', 'L':'0', 'R':'1'})
    board_row = int(board_data[:7], 2)
    board_col = int(board_data[7:10], 2)
    airplane[board_row][board_col] = '#'

while i <= 127 * 8:
    if(row >= 10 and row <= 80 and col <= 7):
        if airplane[row][col] is not '#':
            seat_id = row * 8 + col
            print("[AVAILABLE SEAT]: [Row: {} | Col: {} >> Seat ID: {}]".format(row, col, seat_id))

    if(row % 127 == 0):
        row = 0
        col += 1
    row +=  1
    i += 1

#Result 597

File: 2020_advent/Day_5-[Binary_Boarding]/php/exercise_2.php
<?php

$input = file_get_contents("input_data.txt");

$arr = explode("\r\n", $input);
$airplane = array_fill(0, 127, array_fill(0, 8, "."));

foreach($arr as $board_seat) {
    $board_data = str_replace(['F', 'B', 'L', 'R'], [0, 1, 0, 1], $board_seat);
    $board_data = str_split($board_data, 7);
    $board_seat_row = bindec($board_data[0]);
    $board_seat_col = bindec($board_data[1]);
    $airplane[$board_seat_row][$board_seat_col] = "#";
}

$row = 0;
$col = 0;

for($i = 0; $i <= 127 * 8; ++$i) {

    if(@$airplane[$row][$col] !== "#" && $row >= 10 && $row <= 80 && $col !== 8) {
        $seat_id = $row * 8 + $col;
        echo "[AVAILABLE SEAT]: [Row: $row | Col: $col >> Seat ID: $seat_id]";
    };
    if($row % 127 == 0) {
        $row = 0;
        ++$col;
    }
    ++$row;
}

//Result 597